src/V3Randomize.cpp

```diff
@@ -0,0 +1,280 @@
+        const auto pos = m_enumValueTabMap.find(nodep);
+        if (pos != m_enumValueTabMap.end()) return pos->second;
+        UINFO(9, "Construct Venumvaltab " << nodep << endl);
+        size_t itemCount = 0;  // Number of enum items
+        for (auto* itemp = nodep->itemsp(); itemp; itemp = VN_CAST(itemp->nextp(), EnumItem))
+            itemCount++;
+        AstNodeArrayDType* vardtypep = new AstUnpackArrayDType(
+            nodep->fileline(), nodep->dtypep(), new AstRange(nodep->fileline(), itemCount, 0));
+        AstInitArray* initp = new AstInitArray(nodep->fileline(), vardtypep, nullptr);
+        v3Global.rootp()->typeTablep()->addTypesp(vardtypep);
+        AstVar* varp
+            = new AstVar(nodep->fileline(), AstVarType::MODULETEMP,
+                         "__Venumvaltab_" + cvtToStr(m_enumValueTabMap.size()), vardtypep);
+        varp->isConst(true);
+        varp->isStatic(true);
+        varp->valuep(initp);
+        // Add to root, as don't know module we are in, and aids later structure sharing
+        v3Global.rootp()->dollarUnitPkgAddp()->addStmtp(varp);
+        UASSERT_OBJ(nodep->itemsp(), nodep, "Enum without items");
+        for (AstEnumItem* itemp = nodep->itemsp(); itemp;
+             itemp = VN_CAST(itemp->nextp(), EnumItem)) {
+            AstConst* vconstp = VN_CAST(itemp->valuep(), Const);
+            UASSERT_OBJ(vconstp, nodep, "Enum item without constified value");
+            initp->addValuep(vconstp->cloneTree(false));
+        }
+        m_enumValueTabMap.insert(make_pair(nodep, varp));
+        return varp;
+    }
+    AstNodeStmt* newRandStmtsp(FileLine* fl, AstNodeVarRef* varrefp, int offset = 0,
+                               AstMemberDType* memberp = nullptr) {
+        if (auto* structDtp
+            = VN_CAST(memberp ? memberp->subDTypep()->skipRefp() : varrefp->dtypep()->skipRefp(),
+                      StructDType)) {
+            AstNodeStmt* stmtsp = nullptr;
+            offset += memberp ? memberp->lsb() : 0;
+            for (auto* memberp = structDtp->membersp(); memberp;
+                 memberp = VN_CAST(memberp->nextp(), MemberDType)) {
+                auto* randp = newRandStmtsp(fl, stmtsp ? varrefp->cloneTree(false) : varrefp,
+                                            offset, memberp);
+                if (stmtsp)
+                    stmtsp->addNext(randp);
+                else
+                    stmtsp = randp;
+            }
+            return stmtsp;
+        } else {
+            AstNodeMath* valp;
+            if (auto* enumDtp = VN_CAST(memberp ? memberp->subDTypep()->subDTypep()
+                                                : varrefp->dtypep()->subDTypep(),
+                                        EnumDType)) {
+                size_t itemCount = 0;  // Number of enum items
+                for (auto* itemp = enumDtp->itemsp(); itemp;
+                     itemp = VN_CAST(itemp->nextp(), EnumItem))
+                    itemCount++;
```

Comment:
Same comment as above.  Given this occurs 2 places perhaps better to make it a AstEnumDType member function.